File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    answer = 0
    h = []
    for i in scoville:
        heapq.heappush(h,i)
    while True:
        if h[0] >= K:
            break
        elif len(h) < 2:
            answer = -1
            break
        else:
            a = heapq.heappop(h)
            b = heapq.heappop(h)
            heapq.heappush(h,a+b*2)
            answer +=1
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;
class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        int s = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for (Integer i : scoville) {
            pq.add(i);
        }
        while (true) {
            s = pq.poll();
            if (s >= K) break;
            if (pq.size() < 1) {
                answer = -1;
                break;
            }
            pq.add(s + (pq.poll()*2));
            answer +=1;
        }
        return answer;
    }
}